datafusion/substrait/src/logical_plan/producer.rs

```diff
@@ -1127,6 +1162,48 @@ pub fn to_substrait_rex(
                 })),
             })
         }
+        Expr::InSubquery(InSubquery {
+            expr,
+            subquery,
+            negated,
+        }) => {
+            let substrait_expr =
+                to_substrait_rex(ctx, expr, schema, col_ref_offset, extension_info)?;
+
+            let subquery_plan =
+                to_substrait_rel(subquery.subquery.as_ref(), ctx, extension_info)?;
```

Comment:
I added `ctx` here so that I could call `to_substrait_rel` recursively. All other functions in this file which have an added argument of `ctx` are to support this